### DISCUSSION TITLE
다크모드를 추가하면 좋을 것 같아요 !

### DISCUSSION THREAD

**Discussion by jujubebat** [darass-team/2021-darass#741] [General]
😃

**Comment by zereight** [comment]
다크모드 관련 논의사항이 있습니다.

https://user-images.githubusercontent.com/42544600/137763005-ac263723-fe22-419d-adab-840a5480ae6e.mov

이렇게 구현된 상태인데요. (유저가 다크모드를 토글할 수 있는 ui존재)

아래에서 한가지를 골라야할것같습니다.

1. [현재] 사용자가 알아서 다크모드 상태를 토글하게한다. 그리고 토글한 상태는 localStorage에 저장하여 새로고침시에도 유지한다. (초기값은 무조건 light상태)
2. 스크립트를 삽입하는 관리자가, 스크립트에 darkMode="true" 속성을 넘겨주고 그것을 고정한다. (유저가 변경할 수 있는 ui가 없음, 관리자가 다크모드 풀려면 스크립트코드에서 해당속성직접제거해야함, 그리고 스크립트발급페이지에서 다크모드설정 안내 별도로 필요)
3. 스크립트를 삽입하는 관리자가, 스크립트에 darkMode="true" 속성을 넘겨주고(이게 초기값, 스크립트발급페이지에서 다크모드설정 안내 별도로 필요), 다크모드 토글 ui도 있어서 사용자가 다크모드를 토글할 수 있게한다. (localStorage에 저장하여 새로고침해도 토글된 상태유지)

**Comment by jaeseongDev** [comment]
1번의 경우에는 관리자 입장에서는, 블로그의 전체 컨셉과 통일시켜서 다크모드를 하고 싶어서 토글을 클릭해서 디자인적으로 문제가 없을 수는 있습니다. 
하지만 다른 사용자들의 입장에서는 토글을 클릭하지 않는 이상, 블로그는 다크테마로 꾸며놨는데 댓글 모듈만 다크모드로 되어 있지 않아서 디자인적으로 불편함을 느낄 수 있다고 생각합니다. 

**그래서 2번의 선택지를 통해, 블로그 관리자가 꾸미고 싶은 대로 고정을 하는 것이 UI/UX 측면에서 더 낫지 않나라고 조심스럽게 의견 제시 해봅니다ㅎㅎㅎ**

제가 생각한 개인적인 의견이므로 참고만 해주세요:)

**Reply by zereight** [reply]
의견감사합니다!

**Comment by Sehwan-Jang** [comment]
저는 개인적으로 3번이 끌리네요!
제리의 의견에서 관리자가 디폴트 값을 조정할수 있는 것도 좋지만 방문자도 토글 스위치를 보고 눌러보는것 자체가 하나의 홍보 수단 (흥미 유발을 통해?)이 될수도 있다고 생각해서 굳이 토글버튼은 막을 필욘 없지 않을까 제안해봅니다 ㅎㅎ

**Reply by zereight** [reply]
의견 감사합니다!

**Comment by jujubebat** [comment]
저도 관리자가 설정하도록 하는게 좋다고 생각합니다!
테마 정보를 서버에 저장해서, 관리자 페이지에서 api 호출해서 테마 변경하는 방법도 있을 것 같습니다~!

**Reply by zereight** [reply]
의견 감사합니다!

**Comment by zereight** [comment]
다수결의견에 따라 2번으로 수정하겠습니다!

**Reply by zereight** [reply]
구현 완료.
